src/data.rs

```diff
@@ -0,0 +1,107 @@
+use core::fmt;
+
+use rand::{thread_rng, Rng};
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize)]
+pub(crate) struct Reading {
+    temperature: f64,
+    humidity: f64,
+    pressure: i32,
+    dust_concentration: f64,
+    air_purity: String,
+}
+
+impl Reading {
+    pub(crate) fn new(
+        temperature: f64,
+        humidity: f64,
+        pressure: i32,
+        dust_concentration: f64,
+        air_purity: String,
+    ) -> Reading {
+        Reading {
+            temperature,
+            humidity,
+            pressure,
+            dust_concentration,
+            air_purity,
+        }
+    }
+}
+
+#[derive(Debug, PartialEq)]
+pub(crate) enum AirPurity {
+    Dangerous,
+    High,
+    Low,
+    FreshAir,
+}
+
+impl AirPurity {
+    pub(crate) fn from_value(value: f64) -> AirPurity {
+        match value {
+            value if (f64::MIN..=50.0).contains(&value) => AirPurity::FreshAir,
+            value if value > 50.0 && value <= 100.0 => AirPurity::Low,
+            value if value > 100.0 && value <= 150.0 => AirPurity::High,
+            _ => AirPurity::Dangerous,
+        }
+    }
+}
+
+// implements fmt::Display for AirPurity so that we can call .to_string() on
+// each enum value
+impl fmt::Display for AirPurity {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            AirPurity::Low => write!(f, "Fresh Air"),
+            AirPurity::High => write!(f, "Low Pollution"),
+            AirPurity::Dangerous => write!(f, "High Pollution"),
+            AirPurity::FreshAir => write!(f, "Dangerous Pollution"),
```

Comment:
Should this be `"Fresh Air"` ?

Reply:
Oh nice catch. This is not mine, I just moved it around, you can see it in master [here](https://github.com/Jim-Hodapp-Coaching/ambi_mock_client/blob/9937b55c33869852513441a5c2aee307b2f81ed0/src/lib.rs#L84-L93). What should I name it instead?

Reply:
I'm guessing I added that :)

Actually looking at it now it's all a bit off... probably also my fault

What do you think about?
```rust
AirPurity::Low => write!(f, "Low Pollution"),
AirPurity::High => write!(f, "High Pollution"),
AirPurity::Dangerous => write!(f, "Dangerous Pollution"),
AirPurity::FreshAir => write!(f, "Fresh Air"),
```

Reply:
Makes more sense, will add that!